Products/ZenModel/UserSettings.py

```diff
@@ -1152,6 +1152,31 @@ def manage_deleteAdministrativeRole(self, delids=(), REQUEST=None):
             return self.callZenScreen(REQUEST)
 
 
+    security.declareProtected(ZEN_CHANGE_SETTINGS, 'getAllAdminGuids')
+    def getAllAdminGuids(self, returnChildrenForRootObj=False):
+        """
+        Return all guids user has permissions to.
+        """
+        guids = []
+        rootOrganizers = (
+                        '/zport/dmd/Devices',
+                        '/zport/dmd/Locations',
+                        '/zport/dmd/Groups',
+                        '/zport/dmd/Systems'
+        )
+        ars = self.getAllAdminRoles()
+        if not returnChildrenForRootObj:
+            guids.extend(IGlobalIdentifier(ar.managedObject()).getGUID() for ar in ars)
+        else:
+            for ar in ars:
+                if ar.managedObject().getPrimaryId() in rootOrganizers:
+                    guids.extend(IGlobalIdentifier(child).getGUID() for child in ar.managedObject().children())
+                else:
+                    guids.extend(IGlobalIdentifier(ar.managedObject()).getGUID())
```

Comment:
This one should be `append`, not `extend`; only a single element is added to the list.